docs/security/requirements.mdx

```diff
@@ -0,0 +1,120 @@
+
+## Vault data (VD)
+
+:::warning Draft - Early version
+
+This section is still in its early stages and does not yet reflect current or future standards.
+
+:::
+
+1. Vault data **MUST** be protected _at rest_.
+
+   - a. The Client **MUST** encrypt vault data stored on disk.
+   - b. The Client **MUST** use the UserKey to encrypt vault data stored on the Server.
+   - c. The Client **SHOULD** ensure that no mechanisms exists such that vault data may be stored to
+     disk unencrypted without user consent. This includes cases where it is not the Client itself
+     that stores the data to disk.
+   - d. The Client **MUST** encrypt any vault data derivatives that can be used to re-create the
+     original form or are considered to be vault data on their own.
+   - e. The Client **MUST NOT** store any artifacts (e.g. encryption keys) on disk such that the
+     encrypted vault data can be decrypted without any additional information provided by the User.
+     - i. The Client **MAY** store such artifacts if given an _informed and explicit consent_ by the
+       User.
+   - f. The Client **MUST NOT** store any artifacts (e.g. encryption keys) such that the vault data
+     can be decrypted by the Server, regardless of consent.
+
+2. Vault data **MAY** be unprotected while _in use_.
+
+   - a. The Client **MAY** decrypt all vault data during vault unlock.
+     - i. The Client **SHOULD** minimize the quantity of decrypted vault data.
+   - b. The Client **MAY** leave vault data encrypted after unlock.
+   - c. The Client **MUST** ensure that unprotected data does not linger when no longer _in use_.
+
+3. Vault data **SHOULD** be protected while _in transit_.
+
+   - 1. The Client **MAY** use unprotected transmissions to the display/monitor.
+   - 2. The Client **SHOULD** use a trusted channel when the transmission crosses process
+        boundaries.
+   - 3. The Client **MUST** use a trusted channel if there is a risk that the data can be
+        eavesdropped by unintended parties.
+   - 4. The Client **MUST** certify that the receiver is within the Bitwarden secure environment.
+   - 5. The Client **MUST** use a trusted channel when the transmission crosses device boundaries.
+
+4. Vault data **MUST NOT** be shared without an _informed and explicit consent_ by the User.
+
+   - 1. The Client **MAY** trust the OS to gather consent.
+   - 2. The Client **MAY** augment the OS when the consent process is not sufficiently clear or
+        explicit.
+   - 3. The Client **SHOULD** guarantee that the third party is the receiver.
+
+## Encryption keys (EK)
+
+:::warning Draft - Early version
+
+This section is still in its early stages and does not yet reflect current or future standards.
+
+:::
+
+1. The UserKey **MUST** have 256 bits of security.
+2. The UserKey **MUST** be protected _at rest_.
+3. The UserKey **MAY** be unprotected while _in use_.
+4. The UserKey **MUST** be protected while _in transit_.
+5. The UserKey **MUST NOT** be shared.
```

Comment:
You're actually the second person to give me this exact comment so there is definitely room for improvement here. Account recovery and emergency access is not "sharing" according to the definition: 

> **Data sharing**
> The controlled process in which data leaves the Bitwarden secure environment unprotected. As a consequence the guarantees made by this document will no longer apply. The receiving party may or may not have its own guarantees.

Reply:
Interesting. Given that "sharing" has a common use, maybe there is a more specific term that can be used here? (Maybe "export" - or would that be confused with a vault export?)

Alternatively, it's only used a couple of times - maybe you remove the definition and be explicit each time: "The UserKey MUST NOT leave the Bitwarden secure environment in an unprotected state."

---

As a separate question - does this document have anything to say about when data is exchanged/accessed between users then? EA and Account Recovery are within the Bitwarden secure environment, but it has security implications.

Reply:
Both very good suggestions. I think we might need the definition just to make the document easier to read, but you are right that it might be mixed up with vault export. Still think its worth it though, I'll update it when I get a chance!